### DISCUSSION TITLE
Adding MySQL as an LXC-Container

### DISCUSSION THREAD

**Discussion by andygrunwald** [tteck/Proxmox#3449] [Ideas]
This repository offers MariaDB already, which is nice.
While MariaDB was created as a fork of MySQL a while ago, both projects are developing in different directions right now.

Offering a real MySQL would be a great thing.

**Comment by timothystewart6** [comment]
Please consider this.  MariaDB isn't a drop in replacement for my MySQL. While I wish it were, there are still subtle differences.  The differences are enough to force smaller projects not to test with MariaDB due to resources. 

e.g.

GhostBlog
https://ghost.org/docs/faq/supported-databases/

While "It should just work ™️ "  It does not since MariaDB has made changes that are not compatible with MySQL.

**Comment by timothystewart6** [comment]
I am happy opening a PR for this

**Comment by MickLesk** [comment]
It is not an problem to build a MySQL LXC, the problem is that MySQL is not available in the Debian or Ubuntu repository, which means you would have to download and configure it from an external source. In addition, the repository of mySQL is not really good, you don't see “updates” directly, which means you would always have to use a fixed version in the LXC and manually adjust it again in the script after x months. I don't know if it's worth the time and effort to make an LXC for this. Anyone who can set up a Debian 12 LXC can install MySQL themselves with a few console inputs.

Here is a good example:
https://www.linuxbuzz.com/how-to-install-mysql-on-debian-12/

**Reply by tteck** [reply]
Why not use their repository?
```
curl -fsSL https://repo.mysql.com/RPM-GPG-KEY-mysql-2023 | gpg --dearmor  -o /usr/share/keyrings/mysql.gpg
echo "deb [signed-by=/usr/share/keyrings/mysql.gpg] http://repo.mysql.com/apt/debian $(lsb_release -sc) mysql-8.0" >/etc/apt/sources.list.d/mysql.list
```
after adding the required dependencies (`gpg` & `lsb-release`)

**Reply by timothystewart6** [reply]
I agree, this is what I assumed would need to happen

**Reply by tteck** [reply]
I don't know if `apt-get install -y mysql-community-server` can be run unattended.

**Reply by MickLesk** [reply]
> I don't know if `apt-get install -y mysql-community-server` can be run unattended.

nope, that's exactly what I meant. You have to click/enter/type through the install agent (at least that's the state of my customers' systems at the begin of the year) 

But I can look again tomorrow.

I find their repo really awful :D but if you don't mind, I'll have a look.

**Reply by tteck** [reply]
It can by using `export DEBIAN_FRONTEND=noninteractive`

**Comment by tteck** [comment]
_Should_ be pretty simple.
```
apt-get install -y curl sudo mc gpg lsb-release
curl -fsSL https://repo.mysql.com/RPM-GPG-KEY-mysql-2023 | gpg --dearmor  -o /usr/share/keyrings/mysql.gpg
echo "deb [signed-by=/usr/share/keyrings/mysql.gpg] http://repo.mysql.com/apt/debian $(lsb_release -sc) mysql-8.0" >/etc/apt/sources.list.d/mysql.list
apt-get update
export DEBIAN_FRONTEND=noninteractive
apt-get install -y mysql-community-server
```

Post Install
```
mysql -u root
ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'your_new_password';
FLUSH PRIVILEGES;
exit
mysql_secure_installation
```

**Reply by MickLesk** [reply]
Yep, i prepared an lxc, need little Optimization, tomorrow can i Push it

**Comment by MickLesk** [comment]
@timothystewart6 @andygrunwald can be tested:
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/mysql.sh)"`
